storage/innobase/include/srv0start.h

```diff
@@ -120,7 +121,7 @@ extern bool srv_undo_sources;
 
 /** At a shutdown this value climbs from SRV_SHUTDOWN_NONE to
 SRV_SHUTDOWN_CLEANUP and then to SRV_SHUTDOWN_LAST_PHASE, and so on */
-extern	enum srv_shutdown_t	srv_shutdown_state;
+extern std::atomic<enum srv_shutdown_t> srv_shutdown_state;
```

Comment:
This will break the build on clang:
```
/home/buildbot/amd64-debian-11-msan/build/storage/innobase/fil/fil0crypt.cc:1132:3: error: multiple conversions from switch condition type 'std::atomic<enum srv_shutdown_t>' to an integral or enumeration type
                switch (srv_shutdown_state) {
                ^       ~~~~~~~~~~~~~~~~~~
/usr/lib/llvm-15/bin/../include/c++/v1/atomic:1562:5: note: conversion to enumeration type 'srv_shutdown_t'
```
Furthermore, if we want to change this, I think that we should use `Atomic_relaxed<srv_shutdown_t>` to avoid the implicit default `std::memory_order_seq_cst`, which feels like an overkill to me.

Reply:
While I think it needs careful thoughts to check if such checks should be relaxed atomics I think it is best to keep this part out of the patch. I am now passing the slow_shutdown parameter down as suggested by other comment.